src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -839,7 +860,9 @@ private static PackageDependency GetDependencyVersionRange(LibraryDependency dep
         {
             var range = dependency.LibraryRange.VersionRange;
 
-            if (range == null && dependency.LibraryRange.TypeConstraint == LibraryTypes.ExternalProject)
+            if (range == null
+                && (dependency.LibraryRange.TypeConstraint & LibraryTypeFlag.ExternalProject)
+                    == LibraryTypeFlag.ExternalProject)
```

Comment:
LibraryRange has a helper for AllowsType, I can add a few more there such as AllowsAnyOf for cases where it's checking if it allows a package, project, or external project.